type_helpers.go

```diff
@@ -117,3 +130,138 @@ func SMTPAuthValue(v SMTPAuthType) *SMTPAuthType {
 func String(v string) *string {
 	return &v
 }
+
+// Nullable is a generic type, which implements a field that can be one of three states:
+//
+// - field is not set in the request
+// - field is explicitly set to `null` in the request
+// - field is explicitly set to a valid value in the request
+//
+// Nullable is intended to be used with JSON marshalling and unmarshalling.
+// This is generally useful for PATCH requests, where attributes with zero
+// values are intentionally not marshaled into the request payload so that
+// existing attribute values are not overwritten.
+//
+// Internal implementation details:
+//
+// - map[true]T means a value was provided
+// - map[false]T means an explicit null was provided
+// - nil or zero map means the field was not provided
+//
+// If the field is expected to be optional, add the `omitempty` JSON tags. Do NOT use `*Nullable`!
+//
+// Adapted from https://www.jvt.me/posts/2024/01/09/go-json-nullable/
+
+type Nullable[T any] map[bool]T
+
+// NewNullableWithValue is a convenience helper to allow constructing a
+// Nullable with a given value, for instance to construct a field inside a
+// struct without introducing an intermediate variable.
+func NewNullableWithValue[T any](t T) Nullable[T] {
+	var n Nullable[T]
+	n.Set(t)
+	return n
+}
+
+// NewNullNullable is a convenience helper to allow constructing a Nullable with
+// an explicit `null`, for instance to construct a field inside a struct
+// without introducing an intermediate variable
+func NewNullNullable[T any]() Nullable[T] {
+	var n Nullable[T]
+	n.SetNull()
+	return n
+}
```

Comment:
All these methods/types in `type_helpers` are exported publicly. Is this something we want?